src/core/reactor.cc

```diff
@@ -1959,6 +1959,16 @@ reactor::fstat(int fd) noexcept {
     });
 }
 
+future<size_t> reactor::fill_dirent_buffer(int fd, char* buffer, size_t buffer_size) {
```

Comment:
nit, i'd suggest rename this function to `get_dir_entries()` for better readability, and filling a buffer just but how this function is exposed to its caller, just like how the `read(2)` is exposed to the libc users. i understand that `read(2)` is named "read" for a handful reasons, for instance, its inventor was in favor of very short names back in the old days, but if we compare them we can see why `read()` is a better name than `fill_read_buffer()`.

Reply:
I'd then named it after the syscall -- getdents() or readdir(). WDYT, @tchaikov ?

Reply:
yeah. `readdir()` is better. i'd prefer using a name which is more consistent with other names in reactor. please note, we already have names like

- `make_directory`
- `touch_directory`

so probably `get_directory_entries()` or `read_directory()` is a better alternative.